test/core/VarReplay.jl

```diff
@@ -94,68 +94,20 @@ include("../test_utils/AllUtils.jl")
         randr(vi, vn_z3, dists[1], spl1)
         @test vi.orders == [1, 1, 2, 2, 2, 3]
         @test vi.num_produce == 3
-        @test getretain(vi, spl1) == UnitRange[]
 
-        # Check getretain at different stages
-        # variables samples in different order: z1,a1,z2,z3,a2
         vi.num_produce = 0
-        @test getretain(vi, spl1) == UnitRange[6:6,5:5,4:4,2:2,1:1]
-        @test getretain(vi, spl2) == UnitRange[3:3]
         set_retained_vns_del_by_spl!(vi, spl1)
```

Comment:
Can we test if `is_flagged(vi, vn, "del")` is `true` for the corresponding `vns`?

Reply:
sounds good to adopt these tests for testing `is_flagged(vi, vn, "del")`, since it's important to ensure the functionality adding/deleting variables is correct.

Reply:
Done.